src/net/p2p.rs

```diff
@@ -1284,27 +1508,39 @@ impl PeerNetwork {
             self.rekey(None);
         }
       
-        Ok(unhandled_messages)
+        Ok(network_result)
     }
 
     /// Top-level main-loop circuit to take.
     /// -- polls the peer network state to get new sockets and detect ready sockets
     /// -- carries out network conversations
     /// -- receives and dispatches requests from other threads
-    /// Returns the table of unhandled network messages to be acted upon, keyed by the neighbors
-    /// that sent them.
-    pub fn run(&mut self, burndb: &mut BurnDB, chainstate: &mut StacksChainState, poll_timeout: u64) -> Result<HashMap<usize, Vec<StacksMessage>>, net_error> {
-        let poll_state = match self.network {
+    /// -- runs the http peer main loop
+    /// Returns the table of unhandled p2p network messages to be acted upon, keyed by the neighbors
+    /// that sent them (i.e. keyed by their event IDs)
+    pub fn run(&mut self, burndb: &mut BurnDB, chainstate: &mut StacksChainState, dns_client_opt: Option<&mut DNSClient>, poll_timeout: u64) -> Result<NetworkResult, net_error> {
+        let p2p_poll_state = match self.network {
             None => {
+                test_debug!("{:?}: network not connected", &self.local_peer);
                 Err(net_error::NotConnected)
             },
             Some(ref mut network) => {
                 network.poll(poll_timeout)
             }
         }?;
 
-        let unsolicited_messages = self.dispatch_network(burndb, chainstate, poll_state)?;
-        Ok(unsolicited_messages)
+        let result = self.dispatch_network(burndb, chainstate, dns_client_opt, p2p_poll_state)?;
+        
+        let mut http = self.http.take();
```

Comment:
Does this need to take/replace? `HttpPeer::run` only takes a mutable self reference, rather than ownership.

Reply:
Sure, I can fix that.

Reply:
Fixed in 813927fcdcf82c1240ea31a6990dbcb60631c716

Reply:
Does `.http` need to be an option?

Reply:
Right now, both `.http` and `.network` are options because the act of binding to ports is separate from constructing the struct.  Not exactly adherent to RAII, though; can change if you want.

Reply:
Leaving as an option for that is fine with me